src/services/PatchParser.ts

```diff
@@ -0,0 +1,31 @@
+import {PatchInfo} from "../interfaces/GitHubTypes";
+
+export default class PatchParser {
+
+    private readonly DELIMITER = "\n";
+
+    public extractPatchInfo(name: string, filePatch: string): PatchInfo {
+        let linesAdded: string[] = [];
+        let linesDeleted: string[] = [];
+        if (filePatch !== undefined) {
+            const patchLines: string[] = filePatch.split(this.DELIMITER).filter((line: string) => line.length > 0);
+            patchLines.forEach((line: string) => {
+                return this.populateChangedLineArrays(line, linesAdded, linesDeleted);
+            });
+            return { name: name, linesAdded: linesAdded, linesDeleted: linesDeleted } as PatchInfo;
+        }
+        return {} as PatchInfo;
+    }
+
+    private populateChangedLineArrays(line: string, linesAdded: string[], linesDeleted: string[]) {
+        if (this.isAddition(line)) {
+            return linesAdded.push(line.replace(/[ +]/g, ''));
+        } else {
```

Comment:
Are all lines either linesAdded or linesDeleted? What about lines that are neither added nor deleted? Are they not included in filePatch?

Reply:
The lines that aren't deleted/added aren't included, because they don't provide us any valuable information.